sdk/log.py

```diff
@@ -48,8 +49,39 @@ def print_data(self, data):
             else:
                 print("\t", key, ": ", value)
 
+        if data["status"] == "COMPLETED":
+            self.plot_graphs(data["jobParameters"], batchSteps)
+
     def time_difference(self, start_time, end_time):
         start_date_time_obj = datetime.fromisoformat(start_time.split("+")[0])
         end_date_time_obj = datetime.fromisoformat(end_time.split("+")[0])
         tdelta = end_date_time_obj - start_date_time_obj
         return tdelta.total_seconds()
+
+    def plot_graphs(self, data, batchSteps):
+        plot_data = {}
+        for id, batch_step in batchSteps.items():
+            step_name = batch_step["step_name"]
+            if step_name not in plot_data:
```

Comment:
I would assume this condition is always true no?

Reply:
No, I have added a try except block for that.

Reply:
Yes but I have added try block because if the batch has failed, then the time wont be generated right? In this case is it better to remove the exception handling?

Reply:
Ah thats separate! Dont catch the general exception and hide it b/c that is what you are doing. 
If the client fails to download the data for whatever networking reason than the user needs to know.

Reply:
Not the case where the client fails to download, this is the case where end time is not generated if batch fails. There might be 2 reasons for the exceptions, if the time hasn't generated because the batch has failed or the client didnt download the data due to which size and time will be undefined.

I have added a print in except for cases where the plot was not generated.